src/Units/KaijuActions.java

```diff
@@ -0,0 +1,943 @@
+          {
+            // Apply Ram - you lose its bonus move, but kill the target for free
+            canCounter = false;
+            --kaijuState.movePower;
+          }
+
+          if( canCounter )
+          {
+            counter = victimType.kaijuCounter;
+            counter += KaijuWarsWeapons.getCounterBoost(victim, gameMap, location.getEnvironment().terrainType);
+            if( kaijuState.model.isLandUnit()
+                && victimType.slowsLand )
+              --kaijuState.movePower;
+            if( kaijuState.model.isAirUnit()
+                && victimType.slowsAir )
+              --kaijuState.movePower;
+          }
+        }
+        final boolean isLethal = true;
+
+        ArrayList<Unit> stompable = new ArrayList<>();
+        stompable.add(victim);
+        crushEvents.add(new MassDamageEvent(kaijuState.CO, stompable, stompDamage, isLethal));
+        kaijuState.damageHP(counter);
+        if( stompDamage >= victim.getHP() )
+          Utils.enqueueDeathEvent(victim, crushEvents);
+
+        // If we're Hell Turkey and we'll drop below the landing threshold, land.
+        if( kaijuState.model instanceof HellTurkey &&
+            KaijuWarsKaiju.BIRD_LAND_HP >= kaijuState.getHP() )
+        {
+          HellTurkeyLand devolveToType = ((HellTurkey) kaijuState.model).turkeyLand;
+          crushEvents.add(new TransformLifecycle.TransformEvent(kaijuState.unit, devolveToType));
+          kaijuState.model = devolveToType;
+        }
+
+        crushEvents.add(new MassDamageEvent(victim.CO, Arrays.asList(kaijuState.unit), counter, isLethal));
+        // If there's enough counter damage to kill us, die
+        if( kaijuState.getHP() < 1 )
+        {
+          // Kaiju dies at his current position on the path
+          Utils.enqueueDeathEvent(kaijuState.unit, kaijuState.coord, true, crushEvents);
+          kaijuState.movePower = 0;
+        }
+        else // Recalculate remaining movement so taking damage can drain movement
+        {
+          int moveSpent = startingMP - kaijuState.movePower;
+          kaijuState.calculateMovePower();
+          kaijuState.movePower -= moveSpent;
+        }
+      }
+
+      // Add property damage and movement if I can reach
+      final int distance = kaijuState.coord.getDistance(destCoord);
+      if( kaijuState.movePower >= distance )
+      {
+        if( isCrushable(location) )
+        {
+          Environment oldEnvirons = location.getEnvironment();
+          Environment newEnvirons = Environment.getTile(oldEnvirons.terrainType.getBaseTerrain(), oldEnvirons.weatherType);
+          crushEvents.add(new MapChangeEvent(location.getCoordinates(), newEnvirons));
+          if( stomperType.regenOnBuildingKill )
+            crushEvents.add(new HealUnitEvent(kaijuState.unit, 2, null, true));
+          if( stomperType.chargeOnBuildingKill && location.isCaptureable() )
+            // TODO: This is definitely wrong
+            kaijuState.CO.modifyAbilityPower(1);
```

Comment:
Any plans to change this?

Reply:
The correct fix is to make an event to change `Commander` energy levels.
So yes, but also that's been kicked down the road for quite a while now - I can fix it here, but I dunno if it's in-scope for this PR.

Reply:
This is part of why I started the discussion above about handling gamestate changes in event listeners.
Specifically:
If gamestate changes in listeners are a-ok, then I should add an event for this and leave the `Commander` charge listeners alone until we want to add animations.
If not, then I need to touch `Commander` and should probably hunt down all the rest of the instances and correct them - that sounds like a separate PR to me.

Reply:
Agree not in scope for this PR.

Reply:
Ooooooor I guess we can add an energy event now.

Reply:
Making the energy event go all the places it wants to go isn't in scope, but adding the event itself seems fine.